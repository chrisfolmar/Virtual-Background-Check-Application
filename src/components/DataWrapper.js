import React, { Component } from 'react'
import Data from './Data'
import BadIp from './BadIp'
import GoodIp from './GoodIp'

class DataWrapper extends Component {
    render(){
        const address = this.props.address,
            latitudeLongitude = this.props.latitude + ", " + this.props.longitude,
            found = this.props.found,
            timeZone = this.props.timeZone,
            continent = this.props.continent,
            country = this.props.country,
            region = this.props.region,
            city = this.props.city,
            postal = this.props.postal,
            hostname = this.props.hostname,
            asNumber = this.props.asNumber,
            asName = this.props.asName,
            asCountry = this.props.asCountry;

            if(found){
                return (
                    <section className="features-icons bg-light text-center">
                    <div className="container">

                      <div className="row">
        
                          <BadIp address={address} />

                      </div>

                      <h4>IP Geolocation</h4>
                      <div className="row">

                        <div className="col-lg-3">
                          <Data header="Hostname" data={hostname} />
                        </div>

                        <div className="col-lg-3">
                          <Data header="Continent" data={continent} />
                        </div>

                        <div className="col-lg-3">
                          <Data header="Country" data={country} />
                        </div>

                        <div className="col-lg-3">
                          <Data header="Region" data={region} />
                        </div>
                        
                      </div>
                      <div className="row">

                        <div className="col-lg-3">
                          <Data header="City" data={city} />
                        </div>

                        <div className="col-lg-3">
                          <Data header="Postal Code" data={postal} />
                        </div>

                        <div className="col-lg-3">
                          <Data header="Latitude, Longitude" data={latitudeLongitude} />
                        </div>

                        <div className="col-lg-3">
                          <Data header="Time Zone" data={timeZone} />
                        </div>

                      </div>
                      <h4>Autonomous System</h4>
                      <div className="row">

                        <div className="col-lg-4">
                          <Data header="Time Zone" data={timeZone} />
                        </div>

                        <div className="col-lg-4">
                          <Data header="Name" data={asName} />
                        </div>

                        <div className="col-lg-4">
                          <Data header="Country" data={this.props.asCountry} />
                        </div>
        
                      </div>

                    </div>
                  </section>

            );
        }

        return (
            <section className="features-icons bg-light text-center">
              <div className="container">

                <div className="row">
                    
                    <GoodIp address={address} />
                
                </div>

                <h4>IP Geolocation</h4>
                <div className="row">

                  <div className="col-lg-3">
                    <Data header="Hostname" data={hostname} />
                  </div>

                  <div className="col-lg-3">
                    <Data header="Continent" data={continent} />
                  </div>

                  <div className="col-lg-3">
                    <Data header="Country" data={country} />
                  </div>

                  <div className="col-lg-3">
                    <Data header="Region" data={region} />
                  </div>
                  
                </div>
                <div className="row">

                  <div className="col-lg-3">
                    <Data header="City" data={city} />
                  </div>

                  <div className="col-lg-3">
                    <Data header="Postal Code" data={postal} />
                  </div>

                  <div className="col-lg-3">
                    <Data header="Latitude, Longitude" data={latitudeLongitude} />
                  </div>

                  <div className="col-lg-3">
                    <Data header="Time Zone" data={timeZone} />
                  </div>

                </div>    
                <h4>Autonomous System</h4>
                <div className="row">

                  <div className="col-lg-4">
                    <Data header="Number" data={asNumber} />
                  </div>

                  <div className="col-lg-4">
                    <Data header="Name" data={asName} />
                  </div>

                  <div className="col-lg-4">
                    <Data header="Country" data={asCountry} />
                  </div>
  
                </div>
              </div>
            </section>
        );
    }
}



export default DataWrapper;


