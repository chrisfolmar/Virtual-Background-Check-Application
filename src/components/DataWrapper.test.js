import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import DataWrapper from './DataWrapper';

const found=false,
address="1.2.3.4",
hostname="Test Name",
continent="Test Continent",
country="Test Country",
region="Test Region",
city="Test City",
postal="12345",
longitude=45,
latitude=-72,
timeZone="Test Time Zone",
asNumber=123,
asName="Test AS Name",
asCountry="Test AS Country";

describe('DataWrapper', () => { 

	it('renders fully without crashing', () => {
		const component = mount(
			<DataWrapper
				found={found}
				address={address}
				hostname={hostname}
				continenet={continent}
				country={country}
				region={region}
				city={city}
				postal={postal}
				longitude={longitude} 
				latitude={latitude}
				timeZone={timeZone} 
				asNumber={asNumber}
				asName={asName}
				asCountry={asCountry}
			/>
			);
		component.unmount();
	});

	it('full snapshot rendering', () => {
		const component = mount(
			<DataWrapper
				found={found}
				address={address}
				hostname={hostname}
				continenet={continent}
				country={country}
				region={region}
				city={city}
				postal={postal}
				longitude={longitude} 
				latitude={latitude}
				timeZone={timeZone} 
				asNumber={asNumber}
				asName={asName}
				asCountry={asCountry}
			/>
			);
		expect(mountToJson(component)).toMatchSnapshot();
		component.unmount();
	});

	it('renders child with propogated data', () => {
		const component = mount(
			<DataWrapper
				found={found}
				address={address}
				hostname={hostname}
				continenet={continent}
				country={country}
				region={region}
				city={city}
				postal={postal}
				longitude={longitude} 
				latitude={latitude}
				timeZone={timeZone} 
				asNumber={asNumber}
				asName={asName}
				asCountry={asCountry}
			/>
			);
		const populatedDataComponentElement = <p className="mb-0">{country}</p>;
		expect(component.contains(populatedDataComponentElement)).toEqual(true);
	});

});