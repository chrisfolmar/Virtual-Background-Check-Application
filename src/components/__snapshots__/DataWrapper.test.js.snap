// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataWrapper full snapshot rendering 1`] = `
<DataWrapper
  address="1.2.3.4"
  asCountry="Test AS Country"
  asName="Test AS Name"
  asNumber={123}
  city="Test City"
  continenet="Test Continent"
  country="Test Country"
  found={false}
  hostname="Test Name"
  latitude={-72}
  longitude={45}
  postal="12345"
  region="Test Region"
  timeZone="Test Time Zone"
>
  <section
    className="features-icons bg-light text-center"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <GoodIp
          address="1.2.3.4"
        >
          <div
            className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
          >
            <h4>
              1.2.3.4
            </h4>
            <h5
              style={
                Object {
                  "color": "#06B400",
                }
              }
            >
              Not Found!
            </h5>
            <p
              className="mb-0"
            >
              The IP was not found in the database of blacklists!
            </p>
          </div>
        </GoodIp>
      </div>
      <h4>
        IP Geolocation
      </h4>
      <div
        className="row"
      >
        <div
          className="col-lg-3"
        >
          <Data
            data="Test Name"
            header="Hostname"
            icon="icon-list"
          >
            <div
              className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
            >
              <div
                className="features-icons-icon d-flex"
              >
                <i
                  className="icon-list m-auto text-primary"
                />
              </div>
              <h5>
                Hostname
              </h5>
              <p
                className="mb-0"
              >
                Test Name
              </p>
            </div>
          </Data>
        </div>
        <div
          className="col-lg-3"
        >
          <Data
            header="Continent"
            icon="icon-map"
          >
            <div
              className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
            >
              <div
                className="features-icons-icon d-flex"
              >
                <i
                  className="icon-map m-auto text-primary"
                />
              </div>
              <h5>
                Continent
              </h5>
              <p
                className="mb-0"
              />
            </div>
          </Data>
        </div>
        <div
          className="col-lg-3"
        >
          <Data
            data="Test Country"
            header="Country"
            icon="icon-home"
          >
            <div
              className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
            >
              <div
                className="features-icons-icon d-flex"
              >
                <i
                  className="icon-home m-auto text-primary"
                />
              </div>
              <h5>
                Country
              </h5>
              <p
                className="mb-0"
              >
                Test Country
              </p>
            </div>
          </Data>
        </div>
        <div
          className="col-lg-3"
        >
          <Data
            data="Test Region"
            header="Region"
            icon="icon-direction"
          >
            <div
              className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
            >
              <div
                className="features-icons-icon d-flex"
              >
                <i
                  className="icon-direction m-auto text-primary"
                />
              </div>
              <h5>
                Region
              </h5>
              <p
                className="mb-0"
              >
                Test Region
              </p>
            </div>
          </Data>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-lg-3"
        >
          <Data
            data="Test City"
            header="City"
            icon="icon-vector"
          >
            <div
              className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
            >
              <div
                className="features-icons-icon d-flex"
              >
                <i
                  className="icon-vector m-auto text-primary"
                />
              </div>
              <h5>
                City
              </h5>
              <p
                className="mb-0"
              >
                Test City
              </p>
            </div>
          </Data>
        </div>
        <div
          className="col-lg-3"
        >
          <Data
            data="12345"
            header="Postal Code"
            icon="icon-envelope"
          >
            <div
              className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
            >
              <div
                className="features-icons-icon d-flex"
              >
                <i
                  className="icon-envelope m-auto text-primary"
                />
              </div>
              <h5>
                Postal Code
              </h5>
              <p
                className="mb-0"
              >
                12345
              </p>
            </div>
          </Data>
        </div>
        <div
          className="col-lg-3"
        >
          <Data
            data="-72, 45"
            header="Latitude, Longitude"
            icon="icon-globe"
          >
            <div
              className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
            >
              <div
                className="features-icons-icon d-flex"
              >
                <i
                  className="icon-globe m-auto text-primary"
                />
              </div>
              <h5>
                Latitude, Longitude
              </h5>
              <p
                className="mb-0"
              >
                -72, 45
              </p>
            </div>
          </Data>
        </div>
        <div
          className="col-lg-3"
        >
          <Data
            data="Test Time Zone"
            header="Time Zone"
            icon="icon-clock"
          >
            <div
              className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
            >
              <div
                className="features-icons-icon d-flex"
              >
                <i
                  className="icon-clock m-auto text-primary"
                />
              </div>
              <h5>
                Time Zone
              </h5>
              <p
                className="mb-0"
              >
                Test Time Zone
              </p>
            </div>
          </Data>
        </div>
      </div>
      <h4>
        Autonomous System
      </h4>
      <div
        className="row"
      >
        <div
          className="col-lg-4"
        >
          <Data
            data={123}
            header="Number"
            icon="icon-credit-card"
          >
            <div
              className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
            >
              <div
                className="features-icons-icon d-flex"
              >
                <i
                  className="icon-credit-card m-auto text-primary"
                />
              </div>
              <h5>
                Number
              </h5>
              <p
                className="mb-0"
              >
                123
              </p>
            </div>
          </Data>
        </div>
        <div
          className="col-lg-4"
        >
          <Data
            data="Test AS Name"
            header="Name"
            icon="icon-screen-desktop"
          >
            <div
              className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
            >
              <div
                className="features-icons-icon d-flex"
              >
                <i
                  className="icon-screen-desktop m-auto text-primary"
                />
              </div>
              <h5>
                Name
              </h5>
              <p
                className="mb-0"
              >
                Test AS Name
              </p>
            </div>
          </Data>
        </div>
        <div
          className="col-lg-4"
        >
          <Data
            data="Test AS Country"
            header="Country"
            icon="icon-globe-alt"
          >
            <div
              className="features-icons-item mx-auto mb-5 mb-lg-0 mb-lg-3"
            >
              <div
                className="features-icons-icon d-flex"
              >
                <i
                  className="icon-globe-alt m-auto text-primary"
                />
              </div>
              <h5>
                Country
              </h5>
              <p
                className="mb-0"
              >
                Test AS Country
              </p>
            </div>
          </Data>
        </div>
      </div>
    </div>
  </section>
</DataWrapper>
`;
